cobalt/h5vcc/h5vcc_settings.cc

```diff
@@ -118,6 +118,32 @@ bool H5vccSettings::Set(const std::string& name, SetValueType value) const {
     }
   }
 
+  if (name.compare(network::kProtocolFilterKey) == 0 &&
+      value.IsType<std::string>() &&
+      value.AsType<std::string>().size() < 16384) {
+    std::string raw_json = value.AsType<std::string>();
+    base::Value old_config_json;
+    persistent_settings_->Get(network::kProtocolFilterKey, &old_config_json);
+
+    if (raw_json.empty() && (!old_config_json.is_string() ||
+                             !old_config_json.GetString().empty())) {
+      persistent_settings_->Set(network::kProtocolFilterKey, base::Value());
+      network_module_->SetProtocolFilterUpdatePending();
+      return true;
+    }
+
+    absl::optional<base::Value> old_config =
+        base::JSONReader::Read(old_config_json.GetString());
```

Comment:
The `old_config_json.is_string()` check in line 128 doesn't take effect when `raw_json` isn't empty.  So if `old_config_json.is_string()` can be false, it could be false here too (and may fail the DCHECK inside GetString()).

Reply:
Great and thank you providing me with the subsequent PR.  Sorry that I wasn't aware of it!